proposals/nullable-reference-types-specification.md

```diff
@@ -224,41 +246,95 @@ A `null_conditional_expression` has the null state "maybe null".
 
 ## Cast expressions
 
+If a cast expression `(T)E` invokes a user-defined conversion, then the null state of the expression is the default null state for its type. Otherwise, if `T` is null-yielding (*nullable* or *unknown*) then the null state is "maybe null". Otherwise the null state is the same as the null state of `E`.
+
 ## Await expressions
 
-***More expressions***
+The null state of `await E` is the default null state of its type.
+
+## The `as` operator
+
+An `as` expression has the null state "maybe null".
+
+## The null-coalescing operator
+
+`E1 ?? E2` has the same null state as `E2`
+
+## The conditional operator
+
+The null state of `E1 ? E2 : E3` is "not null" if the null state of both `E2` and `E3` are "not null". Otherwise it is "maybe null".
+
+## Query expressions
+
+The null state of a query expression is the default null state of its type.
+
+## Assignment operators
+
+`E1 = E2` and `E1 op= E2` have the same null state as `E2` after any implicit conversions have been applied.
```

Comment:
Does this mean we're waiting for the spec for conversions to see the spec for a warning produced by an assignment?